src/main.cpp

```diff
@@ -216,6 +216,23 @@ class Curl {
 			cerr << "Raw response was: " << body.str() << endl;
 		}
 	}
+	std::tuple<bool, string> PingEndpoint() {
+	  curl_easy_setopt(curl, CURLOPT_NOBODY, 1L);
+	  CURLcode res = curl_easy_perform(curl);
+	  if (res != CURLE_OK) {
+	    return {false, "Unable to reach the device registration endpoint " + _url + "; err: " + curl_easy_strerror(res)};
+	  }
+	  gint64 code = 0;
+	  CURLcode get_info_res = curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &code);
+	  if (get_info_res != CURLE_OK) {
+	    return {false, "Error while checking the device registration endpoint; err: unable to get curl info: " + string(curl_easy_strerror(get_info_res))};
+	  }
+	  if (code >= 500) {
+	    // 401 is returned under normal circumstances what indicates that the OTA backend is reachable and functional
```

Comment:
i think this will work with the factory-registration-ref server, but I believe it will return an HTTP 400 - https://github.com/foundriesio/factory-registration-ref/blob/2c94a8764d342be9308b1e0a1491864a86fee542/registration_ref/app.py#L59

Reply:
> I believe it will return an HTTP 400

That's why I included "under normal circumstances" phrase.

Reply:
I'd just clarify both. The registration-ref is "normal". It confused me and made me go read the code to make sure I understood this would work